lib/Futures/SharedState.h

```diff
@@ -155,25 +155,23 @@ class SharedState {
     _callback = std::forward<F>(func);
 
     auto state = _state.load(std::memory_order_acquire);
-    while (true) {
```

Comment:
Could you comment on why this seemingly unrelated change here?

Reply:
The `while` loop seems unnecessary here.
In the code before this change, the while loop can only be repeated if one had also hit the `TRI_ASSERT(false);` assertion directly before. That means in maintainer mode, there will be at most a single loop iteration. In release mode however, there could have been multiple loop iterations in theory, because the assert will not trigger.
In reality, execution should never reach the `TRI_ASSERT(false);` location here. We can still leave it in place for debugging, but as it is not supposed to be reached, we don't need to do anything special for release builds, and can still remove the while loop here.